src/streamlink/plugins/nrk.py

```diff
@@ -35,30 +15,73 @@
     )
 })
 
+_playable_schema = validate.Schema({
+    "playable": validate.all(
+        {
+            "assets": validate.all(
+                [{
+                    "url": validate.url(
+                        scheme="http",
+                        path=validate.any(
+                            validate.endswith(".m3u8"),
+                            validate.endswith(".mp3")
+                        ),
+                    ),
+                    "format": validate.all(validate.text),
+                }]
+            ),
+        }
+    ),
+    "statistics": {
+        "luna": {
+            "data": {
+                "title": validate.text,
+            },
+        },
+    },
+})
+
 
 class NRK(Plugin):
+    _psapi_url = 'https://psapi.nrk.no'
+
     @classmethod
     def can_handle_url(self, url):
         return _url_re.match(url)
 
     def _get_streams(self):
-        # Get the stream type from the url (tv/radio).
-        stream_type = _url_re.match(self.url).group(1).upper()
-        cookie = {
-            "NRK_PLAYER_SETTINGS_{0}".format(stream_type): COOKIE_PARAMS
-        }
-
-        # Construct API URL for this program.
-        baseurl = self.session.http.get(self.url, cookies=cookie, schema=_schema)
-        program_id = _id_re.search(self.url).group(1)
+        # Construct manifest URL for this program.
+        program_type, program_id = _id_re.search(self.url).groups()
+        if program_type == 'direkte':
+            manifest_type = 'channel'
+        elif program_type == 'serie':
+            manifest_type = 'program'
+        elif program_type == 'podkast':
+            manifest_type = 'podcast'
+        manifest_url = urljoin(self._psapi_url, "playback/manifest/{0}/{1}".format(manifest_type, program_id))
 
         # Extract media URL.
-        json_url = urljoin(baseurl, "mediaelement/{0}".format(program_id))
-        res = self.session.http.get(json_url, cookies=cookie)
-        media_element = self.session.http.json(res, schema=_mediaelement_schema)
-        media_url = media_element["mediaUrl"]
+        res = self.session.http.get(manifest_url)
+        manifest = self.session.http.json(res, schema=_playable_schema)
+        asset = manifest['playable']['assets'][0]
+
+        # Some streams such as podcasts are not HLS but plain files.
+        if asset['format'] == 'HLS':
+            return HLSStream.parse_variant_playlist(self.session, asset['url'])
+        else:
+            return [(self._get_title(manifest), HTTPStream(self.session, asset['url']))]
 
-        return HLSStream.parse_variant_playlist(self.session, media_url)
+    def _get_title(self, manifest):
+        statistics = manifest.get("statistics")
+        if not statistics:
+            return None
+        luna = statistics.get("luna")
+        if not luna:
+            return None
+        data = luna.get("data")
+        if not data:
+            return None
+        return data.get("title")
```

Comment:
```suggestion
        return manifest.get("statistics", {}).get("luna", {}).get("data", {}).get("title")
```

Reply:
This is probably unnecessary though, as you already have a fully nested schema for the `statistics` object, and if it's missing or not matching, an error will be returned. You'll have to set it to optional in the schema if it's indeed optional. To do this, set the object to to `validate.optional("statistics")`.

Reply:
I looked at this in more detail and it appears that the `luna` object can either be `null` or missing entirely. The `statistics` object is always present however, so I went with a middle ground here.